client/app/views/steps/welcome.coffee

```diff
@@ -11,9 +25,29 @@ module.exports = class WelcomeView extends StepView
     events:
         'click @ui.next': 'onSubmit'
 
+
     onSubmit: (event) ->
         event.preventDefault()
-        @model.submit()
+        @model.submit() unless @isDisabled()
+        @setDisabled()
+
+
+    setDisabled: ($el = @ui.next) ->
+        if $el[0].tagName.toLowerCase() in FORMS_DIS_ELEMENTS
+            $el
+                .prop 'disabled', true
+                .attr 'aria-busy', true
+        else
+            $el.attr
+                'aria-disabled': true
+                'aria-busy':     true
+
+
+    isDisabled: ($el = @ui.next) ->
+        if $el[0].tagName.toLowerCase() in FORMS_DIS_ELEMENTS
```

Comment:
I suggest to not trust the DOM and use a local property here like `@isDisabled`. What do you think ?